ext/drpttracer/drpttracer.dox

```diff
@@ -0,0 +1,128 @@
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/**
+***************************************************************************
+***************************************************************************
+\page page_drpttracer Intel PT Tracing
+
+The \p drpttracer DynamoRIO Extension provides clients with tracing functionality via
+Intel's PT instruction tracing feature.
+
+\note This extension only works on x86_64 Linux.
+
+ - \ref sec_drpttracer_setup
+ - \ref sec_drpttracer_usage
+ - \ref sec_drpttracer_tracing_mode
+
+\section sec_drpttracer_setup Setup
+
+To use \p drpttracer with your client simply include this line in your client's
+\p CMakeLists.txt file:
+
+\code use_DynamoRIO_extension(clientname drpttracer) \endcode
+
+That will automatically set up the include path and library dependence.
+
+Initialize and clean up \p drpttracer by calling drpttracer_init() and
+drpttracer_exit().
+
+\section sec_drpttracer_usage Usage
+
+Intel PT (Processor Trace) only logs control flow changes. Therefore, when decoding a
+trace, the decoder needs to get raw bits of every instruction from images.
+\p drpttracer provides APIs to trace PT and generate trace data and some auxiliary data to
+help the decoder to decode the trace. One type of auxiliary data is sideband data. This
+sideband data stores perf event records, which contain the image change message necessary
+for decoding the trace. Another type of auxiliary data is the metadata for a trace. The
+metadata contains the CPU information and some other information that can be used to
+synthesize the time of PT trace and the perf event records. The auxiliary data enables the
+decoder to find the correct raw bits for every instruction.
+
+\p drpttracer provides one data structure to store all generated data:
+
+ - #drpttracer_output_t
+
+And \p drpttracer provides two APIs for client to start and stop tracing:
+
+ - \p drpttracer_start_tracing()
+ - \p drpttracer_end_tracing()
+
+To record a PT trace, the client need use \p drpttracer_start_tracing() to start the
+tracing. The start function let the client specify the following parameters:
+
+ - trace_mode : The tracing mode.
+
+ - pt_size_shift: The size shift of PT trace's ring buffer. It must be greater than 0,
+and the buffer size is 2^pt_size_shift * PAGE_SIZE.
+
+ - sideband_size_shift: The size shift of PT sideband data's ring buffer. It must be
+greater than 0, and the buffer size is 2^sideband_size_shift * PAGE_SIZE.
+
+And the start function will generate a handle for the PT tracing. The handle is a pointer
+to a \p drpttracer's internal data structure pttracer_handle_t, which contains the
+tracing's PT data buffer, sideband data buffer, and metadata.
+
+After tracing is done, the client needs to pass the trace handle to \p
```

Comment:
s/After tracing is done/To stop collecting the trace/

and remove "to stop the tracing" from the end of this sentence.